File: src/ros2_bridge/mappers/std_msgs/int16.rs
//! Typed mapper for `std_msgs/msg/Int16`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn int16_to_bus(msg: ros_env::std_msgs::msg::Int16) -> crate::std_msgs::msg::v1::Int16 {
    crate::std_msgs::msg::v1::Int16 {
        data: msg.data as i32,
    }
}

pub(crate) fn int16_to_ros(bus: crate::std_msgs::msg::v1::Int16) -> ros_env::std_msgs::msg::Int16 {
    ros_env::std_msgs::msg::Int16 {
        data: bus.data as i16,
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct StdMsgsInt16Mapper;

impl TypedTopicMapper for StdMsgsInt16Mapper {
    type Ros = ros_env::std_msgs::msg::Int16;
    type Bus = crate::std_msgs::msg::v1::Int16;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(int16_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(int16_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/std_msgs/int16.py
"""Generated mapper for `std_msgs/msg/Int16`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert


def int16_to_bus(msg):
    from robot_bus.std_msgs.msg.v1 import Int16 as BusMsg

    bus = BusMsg()
    bus.data = int(msg.data)
    return bus


def int16_to_ros(bus):
    from std_msgs.msg import Int16 as RosMsg

    out = RosMsg()
    out.data = int(bus.data)
    return out


class StdMsgsInt16Mapper:
    def type_name(self) -> str:
        return "std_msgs/msg/Int16"

    def ros_msg_type(self):
        from std_msgs.msg import Int16 as RosMsg

        return RosMsg

    def ros_to_bus(self, msg) -> bytes:
        return int16_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        from robot_bus.std_msgs.msg.v1 import Int16 as BusMsg

        bus = BusMsg()
        bus.ParseFromString(payload)
        return int16_to_ros(bus)
